cfApplication.go

```diff
@@ -37,21 +43,28 @@ type Metadata struct {
 // Routes represents a slice of Routes
 type Routes []Route
 
-// Route captures the key elements that define a Route: hostname, protocol and port. These values
+// Route captures the key elements that define a Route in a string that maps to a URL structure. These values
 // are captured as runtime routes, meaning that if the CF Application manifest is configured to disable all routes
 // with the `no-route` value, it will translate into an empty slice.
-// Unless specified by the process or setting the application field `no-route` to true,
-// by default CloudFoundry will always attempt to create a route for each application.
+// By default CloudFoundry will always attempt to create a route for each application, unless specified by the field `no-route` when true
 // For further details check: https://docs.cloudfoundry.org/devguide/deploy-apps/manifest-attributes.html#no-route
 // and https://docs.cloudfoundry.org/devguide/deploy-apps/manifest-attributes.html#random-route
+// Example
+// ---
+//
+//	...
+//	routes:
+//	- route: example.com
+//	  protocol: http2
+//	- route: www.example.com/foo
+//	- route: tcp-example.com:1234
```

Comment:
nit: extra newline?

And, why do we need this example?

Reply:
I just wanted to leave a record of the complexity in which the route source can come from. Otherwise it might seem like overkill to capture the route as a URL, when a simple string with a hostname would suffice. 
I was debating between splitting hostname, path and port or having a single field with an URL. At the end it seemed simple to have the URL field.